gradle-baseline-java/src/main/java/com/palantir/baseline/tasks/CheckJUnitDependencies.java

```diff
@@ -49,178 +42,122 @@ public CheckJUnitDependencies() {
-            if (!engines.contains("jqwik")) {
-                throw new IllegalStateException("Some tests mention jqwik, but the '"
+            if (usesJUnit5 && !hasJunitJupiter) {
+                throw new IllegalStateException("Some tests use JUnit5, but the '"
                         + task.getName()
-                        + "' task does not have the jqwik engine enabled. "
-                        + "This means tests may be silently not running! Please "
-                        + "add the following:\n\n"
-                        + "test {\n"
-                        + "    useJUnitPlatform {\n"
-                        + "        includeEngines 'jqwik', 'junit-jupiter'\n"
-                        + "    }\n"
-                        + "}\n\nAs well as a dependency on the engine:\n\n"
+                        + "' task is not using the JUnit Jupiter engine."
+                        + " To ensure your JUnit5 tests are run, add the following dependency:\n\n"
                         + "    "
-                        + runtime
-                        + " 'net.jqwik:jqwik-engine'\n");
+                        + ss.getRuntimeOnlyConfigurationName()
+                        + " 'org.junit.jupiter:junit-jupiter'\n\n");
             }
-            Preconditions.checkState(
-                    hasDep(deps, CheckJUnitDependencies::isJqwikEngine),
-                    "Some tests mention jqwik, but the '"
-                            + task.getName()
-                            + "' task does not depend on the jqwik engine. "
-                            + "This means tests may be silently not running! Please add the following:\n\n"
-                            + "    "
-                            + runtime
-                            + " 'net.jqwik:jqwik-engine'\n");
-        }
 
-        // spock uses JUnit4 under the hood, so the vintage engine is critical
-        if (spock1Dependency && junitPlatformEnabled) {
-            Preconditions.checkState(
-                    vintageEngineExists,
-                    "Tests may be silently not running! Spock 1.x dependency detected (which uses "
-                            + "a JUnit4 Runner under the hood). Please add the following:\n\n"
-                            + "    "
-                            + testRuntimeOnly
-                            + " 'org.junit.vintage:junit-vintage-engine'\n\n");
-        }
+            if (usesJqwik && !hasJqwikEngine) {
+                throw new IllegalStateException("Some tests use Jqwik, but the '"
+                        + task.getName()
+                        + "' task is not using the Jqwik engine."
+                        + " To ensure your Jqwik tests are run, add the following dependency:\n\n"
+                        + "    "
+                        + ss.getRuntimeOnlyConfigurationName()
+                        + " 'net.jqwik:jqwik-engine'\n\n");
+            }
+        } else {
```

Comment:
In practice, we should never reach this unless someone has explicitly configured a test suite to use something other than JUnit Platform. I don't think we care to support any other test frameworks (all the thing we we care about integrate with JUnit Platform using a custom engine), so I'm tempted to just remove this.